### PR TITLE
add ability to change capacity in fsinfo

### PR DESCRIPTION
**Pull request by h0hmj**
1. add mds rpc
2. add tools-v2 command

### CODE REVIEW

File: curvefs/src/mds/fs_manager.cpp [hunk 1]
```diff
@@ -790,6 +790,39 @@ FSStatusCode FsManager::GetFsInfo(const std::string& fsName, uint32_t fsId,
...
+        LOG(WARNING) << "UpdateFsInfo fs is not inited, fsName = " << fsName;
+        return FSStatusCode::NOT_INITED;
+    } else if (status == FsStatus::DELETING) {
+        LOG(WARNING) << "UpdateFsInfo fs is in deleting, fsName = " << fsName;
+        return FSStatusCode::UNDER_DELETING;
+    }
+
+    // 3. update fs info
+    // set capacity
+    if (req->has_capacity()) {
```
Comment (SeanHai) [inline curvefs/src/mds/fs_manager.cpp:819]: check this at beginning, the step 1 and 2 are not required if capacity not exist.

**Comment by SeanHai** [comment]
Need to add some unit tests

File: curvefs/src/mds/fs_manager.cpp (hunk 1 above)
Comment (h0hmj) [reply curvefs/src/mds/fs_manager.cpp:819]: ok i will add parameter check in MdsServiceImpl::UpdateFsInfo

**Comment by h0hmj** [comment]
> Need to add some unit tests

done

**Comment by h0hmj** [comment]
cicheck

File: curvefs/src/mds/mds_service.cpp [hunk 2]
```diff
@@ -227,6 +227,40 @@ void MdsServiceImpl::GetFsInfo(::google::protobuf::RpcController *controller,
...
+    ::google::protobuf::Closure* done) {
+    (void)controller;
+    brpc::ClosureGuard doneGuard(done);
+
+    LOG(INFO) << "UpdateFsInfo request: " << request->ShortDebugString();
+
+    // request parameter check, we want at least one field that need to change
+    if (!request->has_capacity()) {
+        response->set_statuscode(FSStatusCode::PARAM_ERROR);
+        LOG(ERROR) << "UpdateFsInfo fail, request: "
```
Comment (SeanHai) [inline curvefs/src/mds/mds_service.cpp:243]: log warnning is enough

Comment (h0hmj) [reply curvefs/src/mds/mds_service.cpp]: fix

**Review by SeanHai — APPROVED** [review]

**Comment by h0hmj** [comment]
cicheck

File: curvefs/src/mds/fs_manager.cpp (hunk 1 above)
Comment (Cyber-SiKu) [reply curvefs/src/mds/fs_manager.cpp:819]: It has been checked in [mds servers](https://github.com/h0hmj/curve/blob/d5fe7de44b0b6743210ba5579c1b981567cd3aae/curvefs/src/mds/mds_service.cpp#L240). Do I still need to check here?

File: tools-v2/pkg/cli/command/curvefs/update/fs/fs.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+func (fCmd *FsCommand) AddFlags() {
+	config.AddRpcRetryTimesFlag(fCmd.Cmd)
+	config.AddRpcTimeoutFlag(fCmd.Cmd)
+	config.AddFsMdsAddrFlag(fCmd.Cmd)
+	fCmd.Cmd.Flags().String("fsName", "", "curvefs fsname")
+	fCmd.Cmd.MarkFlagRequired("fsName")
+	// things can be changed
+	fCmd.Cmd.Flags().String("capacity", "", "curvefs capacity")
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvefs/update/fs/fs.go:82]: Add a function like above.

Comment (Cyber-SiKu) [reply tools-v2/pkg/cli/command/curvefs/update/fs/fs.go:82]: just like:
https://github.com/h0hmj/curve/blob/7d8fb7023767386f3dfd0ce3702f4334e7235918/tools-v2/pkg/config/fs.go#L806-L809

File: tools-v2/pkg/cli/command/curvefs/update/fs/fs.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+func (fCmd *FsCommand) Init(cmd *cobra.Command, args []string) error {
+	// args check
+	fsName, _ := cmd.Flags().GetString("fsName")
+	request := &mds.UpdateFsInfoRequest{
+		FsName: &fsName,
+	}
+
+	nothingToChange := true
+	var capacity string
+	if cmd.Flags().Changed("capacity") {
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvefs/update/fs/fs.go:94]:
```suggestion
	if config.GetFlagChanged(ccmd, config.CURVEFS_CAPACITY)
	if cmd.Flags().Changed("capacity") {
```

File: tools-v2/pkg/cli/command/curvefs/update/fs/fs.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+		nothingToChange = false
+		capacity, _ = cmd.Flags().GetString("capacity")
```
Comment (Cyber-SiKu) [inline tools-v2/pkg/cli/command/curvefs/update/fs/fs.go:96]:
```suggestion
		capacity = config.GetStringFlag(cmd, config.CURVEFS_CAPACITY)
```

Comment (h0hmj) [reply tools-v2/pkg/cli/command/curvefs/update/fs/fs.go]: fix

File: tools-v2/pkg/cli/command/curvefs/update/fs/fs.go (hunk 4 above)
Comment (h0hmj) [reply tools-v2/pkg/cli/command/curvefs/update/fs/fs.go]: fix

File: tools-v2/pkg/cli/command/curvefs/update/fs/fs.go (hunk 3 above)
Comment (h0hmj) [reply tools-v2/pkg/cli/command/curvefs/update/fs/fs.go]: fix

**Comment by h0hmj** [comment]
cicheck

**Comment by h0hmj** [comment]
cicheck

**Review by Cyber-SiKu — APPROVED** [review]

**Comment by h0hmj** [comment]
cicheck

**Pull request merged**